build-tools/packages/build-cli/src/library/writeFileIfContentsDiffers.ts

```diff
@@ -0,0 +1,27 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { readFile, writeFile } from "node:fs/promises";
+import { type PathLike } from "node:fs";
+
+/**
+ * Writes to a file, but first reads the contents to check if it matches the desired content. If it does, the operation
+ * is skipped.
+ *
+ * @param filePath - The path to the file to write.
+ * @param contents - The contents to write to the file.
+ * @returns True if the file was written; false otherwise.
+ */
+export async function writeFileIfContentsDiffers(
+	filePath: PathLike,
+	contents: string,
+): Promise<boolean> {
+	const fileContents = await readFile(filePath, { encoding: "utf8" });
+	if (fileContents !== contents) {
```

Comment:
Any concerns about `\n` versus `\r\n` differences here? (I feel like we have issues with consistency there. Maybe doesn't impact the type test case, but might be a problem for general use.)

Reply:
Latest normalizes to LF and clarifies that the function can't be used to output CRLF.